src/devices/machine/t10mmc.cpp

```diff
@@ -224,8 +224,14 @@ void t10mmc::ExecCommand()
 		m_lba = command[2]<<24 | command[3]<<16 | command[4]<<8 | command[5];
 		m_blocks = SCSILengthFromUINT16( &command[7] );
 
-		// special cases: lba of 0 means MSF of 00:02:00
-		if (m_lba == 0)
+		if (m_lba == 0 && cdrom_get_track_type(m_cdrom, 0) == CD_TRACK_AUDIO)
+		{
+			// A request for LBA 0 will return something different depending on the type of media being played.
+			// For data and mixed media, LBA 0 is assigned to MSF 00:02:00 (= LBA 150).
+			// For audio media, LBA 0 is assigned to the actual starting address of track 1.
+			m_lba = cdrom_get_track_start(m_cdrom, 0);
+		}
+		else if (m_lba == 0)
 		{
 			m_lba = 150;
 		}
```

Comment:
I know the optimiser is likely to work it out anyway, but for us humans, could you restructure it to make it clear that both special-cases are for LBA 0?
```
    if (m_lba == 0)
    {
        if (cdrom_get_track_type(m_cdrom, 0) == CD_TRACK_AUDIO)
            ...
        else
            ...
    }
```

Same for the other ones, too.

Reply:
Done.